Fix the list output after a "snappy update"

mvo5 opened the pull request:
After running "snappy update" with an all-snap system the output
is often incorrect. The reboot notification is not displayed, the
date of the install is wrong etc.

The reason for this is that the Update{,All}() calls return a list
of applied updates that is then displayed. However the list of
applied updates is for the remote snaps. We need to convert the
remote snaps to the locally installed ones instead to get the
correct date and the correct reboot-required checks.

elopio commented on snappy/install.go:
```diff
@@ -60,6 +60,26 @@ func doUpdate(part Part, flags InstallFlags, meter progress.Meter) error {
 	return nil
 }
 
+// convertToInstalledSnaps takes a slcie of remote snaps that got
```
typo: slice.

mvo5 replied:
Thanks! Fixed.

elopio commented:
We are starting a pattern here ^_^.
I would like to track bugs in all the features that are published. In my opinion, if it landed in master it's a published feature; so I would like to see a bug in launchpad for all the problems found in master.
In the case of all-snaps, it's arguable if it has been released or not. I'm inclined to say "please report the bug in launchpad", but I'm open to discussion.

elopio commented on snappy/install.go:
```diff
@@ -60,6 +60,26 @@ func doUpdate(part Part, flags InstallFlags, meter progress.Meter) error {
 }
 
+// convertToInstalledSnaps takes a slcie of remote snaps that got
+// updated and finds the coresponding local snap parts and returns them
```
nit: I think this reads better like:
... updated and returns the corresponding local snap parts.

Also you are missing an 'r' in corresponding.

mvo5 replied:
Thanks! Fixed.

elopio commented on snappy/install.go:
```diff
@@ -60,6 +60,26 @@ func doUpdate(part Part, flags InstallFlags, meter progress.Meter) error {
+	for _, part := range remoteUpdates {
+		localPart := FindSnapsByNameAndVersion(part.Name(), part.Version(), installed)
+		if len(localPart) != 1 {
+			return nil, fmt.Errorf("expected one local part for the update %v got %v", part, len(localPart))
```
I would put a comma in here.

"expected one local part for the update %v, got %v"

mvo5 commented:
@elopio Thanks for your review! I addressed your comments, please let me know if there is anything else that needs to be done.

I also created a LP bug for this https://bugs.launchpad.net/snappy/+bug/1531421 here. It sounds like we should discuss the general practise here in a wider form. I personally dislike to have to go to two place for a pull-request. I would prefer to have all the relevant bug data in here instead to avoid the extra work for me to enter it and for the reviewer(s) to look at a different place. But I'm open of course to talk about the pros/cons here.

niemeyer commented:
I really hope that in the refactorings we're pushing we'll statically prevent that sort of bug from happening again. Can you please just add a TODO before convertToInstalledSnaps saying that we want to get rid of it?

LGTM as a temporary fix. Not merging because it sounds like you addressed some concerns by @elopio that are not yet pushed, and it's also missing a second review. @chipaca?

mvo5 commented:
@niemeyer Thanks! I added a note into the branch about the plan for the future. Plus I pushed the fixes for @elopio

elopio commented:
:+1:, and happy that in the future it will be nicer.

elopio commented:
@mvo5 oh, and about the bugs, yes, lets discuss in the next team meeting.
What you propose will only work for us if everybody follows your practice of tagging the branches with 'bugfix'. What I need is a list of the bugs fixed in a release. And what I've found is that it's really hard to dig for that list when it is hidden in a ton of other pull requests. With launchpad, we get that list easily. We could get it without launchpad too, so I'm open to alternatives.
